### ISSUE TITLE
Cut tag warnings displaying incorrectly on Reading page

### ISSUE THREAD

**Issue by zarhooie** [dreamwidth/dw-free#1387]
http://www.dreamwidth.org/support/see_request?id=30867
Original Bugzilla bug: 4070

Screenshot: http://smg.photobucket.com/user/LdyMusyc/media/random/bugsquishing/musycdreamwidthorg2011-11-2011-31-59.png.html

On a personal entry flagged NSFW, the cut tag warning listed as expected with "(You're about to view content which the journal owner has advised should be viewed with discretion.)". But a personal entry flagged 18+ has the cut tag warning as "(You're about to view content that a community administrator has marked as possibly inappropriate for anyone under the age of 18.)". It was definitely a journal, not a community.

Also, a community flagged NSFW has "(You're about to view content which the journal owner has advised should be viewed with discretion.)" on its cut tag.

Basically, they're all flipped around and wrong. Another req for the same issue (possibly useful info in the ICs on that): http://www.dreamwidth.org/support/see_request?id=11206

From 90D: I think the message is wrong when a journal owner has flagged their entire journal instead of the one entry: because journal is associated to 'community adminstrator' that's what the message says instead of 'journal owner'. The other messages were correct for me depending on who flagged the entry (poster or community admin).

**Comment by rahaeli** [comment]
Hm, this sounds familiar. I think it might have been in Bugzilla before the Great Crashening!

**Comment by zarhooie** [comment]
Bug 4070, if anyone has archive access.

**Comment by kaberett** [comment]
claimed :-)

**Comment by zarhooie** [comment]
##from support

**Comment by zarhooie** [comment]
##fromsupport

**Comment by zarhooie** [comment]
##from: support

**Comment by sgsabbage** [comment]
@kaberett has asked me to take a look at this one. I have a fix for it (one of the messages is wrong in en.dat), but then realised we end up in a slightly strange position.

On a personal journal, the message changes depending on whether the whole journal is flagged or just a post. So, if the entire journal is flagged and a post is made at journal default, another user will get this message:

'You're about to view content which the journal owner has advised should be viewed with discretion.'

If only a single post is flagged, the user will get this message:

'You're about to view content which the poster has advised should be viewed with discretion.'

A similar thing happens with 18+. I'm unsure what the behaviour should be here, just the journal owner message?

The problem with communities is pretty much the same, the answer is clearer cut I think, but for completeness and confirmation. If the journal is flagged:

'You're about to view content which the journal owner has advised should be viewed with discretion.'

If a post is flagged by a community admin:

'You are about to view content that a community administrator has advised should be viewed with discretion.'

And if a post is flagged by the poster:

'You're about to view content which the poster has advised should be viewed with discretion.'

I am assuming that in this case the journal flagged message should be the same as the community admin message?

Summary of questions:
Should personal journals just display 'journal owner' rather than 'poster' regardless of how it's flagged?
Should community journals display 'community admin' rather than 'journal owner' if the entire journal is flagged.

Also, can an identity or feed (syndicated?) journal be marked NSFW or 18+?

**Comment by rahaeli** [comment]
1) Yes, personal journals should always use 'journal owner' rather than 'poster' in all messages -- the poster/community administrator distinction is just useful for comms, since a comm admin can override the poster's flagging choice and flag something the poster didn't flag, and we want to preserve that information on whodunit in the messages.

2) Yeah, if the whole journal is flagged (rather than the specific post), let's go with 'community admin'. (I feel like there's room to rewrite those whole-journal-flagged messages to be clearer, but that's outside the scope of this bug and something I can do on-site later.)

3) Identity accounts and feed accounts can never be flagged NSFW or 18+, correct!

**Comment by sgsabbage** [comment]
Awesome, thanks :)

That leaves one final question in regards to 2 then. While the content of the message is the same, do you want to be able to edit the messages to be different between journal owner and community admin?

**Comment by rahaeli** [comment]
Oh, I thought they were already separated! Yes, please: it's functionally the same, I guess, but the distinction between journal owner/community admin is probably worth getting in front of the user.

**Comment by sgsabbage** [comment]
Sorry, I wasn't being clear, they are separated, but I was considering putting them in one string for expediency, but that's easy enough not to do! Thanks :)
